brayns/common/PropertyMap.h

```diff
@@ -74,37 +74,63 @@ class PropertyMap
         }
 
         template <typename T>
-        void assert_valid_enum_type() const
+        void assertValidEnumType() const
         {
             static_assert(std::is_same<T, int32_t>::value ||
                               std::is_same<T, std::string>::value,
                           "Invalid enum type.");
         }
 
+        template <typename T>
+        Type getType() const
```

Comment:
Well, I could also leave it like this and change it to 'if constexpr' when we upgrade to c++17. I wonder though if the compiler optimizes away all if cases since they are known to be false at compile time.

Reply:
It actually does: https://godbolt.org/z/RFcIPK

Reply:
Good to know